frontend/src/components/interaction-manager/content/transaction.tsx

```diff
@@ -119,34 +119,22 @@ export const Transaction = ({
           {description} from <pre>{transaction.wallet}</pre>.
         </p>
       </div>
-      <div style={{ padding: '0 20px 20px' }}>
-        <Title style={{ margin: '0 0 6px' }}>Wallet</Title>
-        <p>{transaction.wallet}</p>
-      </div>
-      <PublicKey publicKey={transaction.publicKey} />
-      <div style={{ padding: '20px 20px 0' }}>
-        <Title style={{ margin: '0 0 12px' }}>Transaction details</Title>
-        <CodeBlock style={{ fontSize: 12, marginBottom: 0 }}>
-          <pre data-testid='transaction-payload'>
-            {JSON.stringify(transaction.payload, null, 2)}
-          </pre>
-        </CodeBlock>
-      </div>
-      <div style={{ padding: '20px 20px 0' }}>
-        <Title style={{ margin: '0 0 6px' }}>Received</Title>
-        <p>{formatDate(new Date(transaction.receivedAt))}</p>
-      </div>
+      <TransactionDetails transaction={transaction} />
       <div style={{ padding: 20 }}>
         <ButtonGroup>
           <Button
-            data-testid='wallet-transaction-request-approve'
+            data-testid='transaction-request-approve'
             onClick={() => onReponse(true)}
+            disabled={status === 'rejecting'}
```

Comment:
question (non blocking): do we want to disable when rejecting OR approving to avoid accidental spam clicks? Same for the reject button below

Reply:
the reject / approve button have a `loading=true` state when they've been clicked (which is also unclickable), and a `disabled={true}` when another action has been triggered